roles/docker/tasks/main.yml

```diff
@@ -96,7 +96,8 @@
       reg_fact_val: "{{ docker_no_proxy | default('') | join(',') }}"
   notify:
     - restart docker
-  when: "{{ 'http_proxy' in openshift.common or 'https_proxy' in openshift.common and docker_check.stat.isreg }}"
+  when:
+    - (docker_check.stat.isreg is defined and docker_check.stat.isreg) and "{{ 'http_proxy' in openshift.common or 'https_proxy' in openshift.common }}"
```

Comment:
that said, my brain has now entered PTO-mode, so I can't remember if the first suggestion can just be wrapped in double quotes or if it does indeed need to be an explicit template.